hooks/shapediver/viewer/interaction/selection/useSelection.ts

```diff
@@ -53,18 +61,26 @@ export function useSelection(
 	for (const outputId in outputs) {
 		// add interaction data
 		if (!patterns[outputId]) patterns[outputId] = [];
-		const { outputNode } = useNodeInteractionData(sessionId, outputId, patterns[outputId], interactionSettings, selectManager);
+		const { outputNode, availableNodeNames: availableNodeNamesForOutput } = useNodeInteractionData(sessionId, outputId, patterns[outputId], interactionSettings, selectManager);
 		// in case selection becomes active or the output node changes, restore the selection status
 		useEffect(() => {
 			if (outputNode && selectManager)
 				restoreSelection(outputNode, selectManager, selectedNodeNames);
 		}, [outputNode, selectManager]);
+
+		// update the available node names
+		useEffect(() => {
+			setAvailableNodeNames(prev => {
+				return { ...prev, [outputs[outputId].name]: availableNodeNamesForOutput };
+			});
+		}, [availableNodeNamesForOutput]);
```

Comment:
@MajorMeerkatThe3rd here is a potential problem: If I understood correctly, `availableNodeNames` never gets reset. I suggest this: 
   * Inside the loop, collect the data from `availableNodeNamesForOutput` into an array `nodeNames` that is not stateful (no useEffect)
   * After the loop, use an useEffect hook to compare the content of `nodeNames` with `availableNodeNames`, call `setAvailableNodeNames` if they are different. 
 
This will also minimizes the number of re-renders

Reply:
@snabela I tried the approach you suggest, but this ends in an endless loop. I also tried some other approaches, but they are have the same number of calls that are used. 

I think this is something that should be done in a general refactoring, of the for loop, which is probably a bigger task.